app/lib/bk/compat/parsers/bitrise/loader.rb

```diff
@@ -0,0 +1,39 @@
+# frozen_string_literal: true
+
+module BK
+  module Compat
+    # Bitrise workflow configuration loader
+    class Bitrise
+      STEP_KEY_REGEX = /([a-zA-Z-]*)@(\d*|0)(\.((\d*)|0)){0,3}/
+
+      def load_workflow(key, config)
+        raise "Duplicate workflow name: #{key}" if @steps_by_key.include?(key)
+
+        @steps_by_key[key] = load_workflow_steps(key, config)
+      end
+
+      def load_workflow_steps(key, config)
+        # Extend for other types of steps - approvals/waits for example.
+        BK::Compat::CommandStep.new(label: key, key: key).tap do |cmd_step|
+          config['steps'].each do |step|
+            step_key, step_config = step.first
+            step_inputs = step_config['inputs'].reduce({}, :merge)
+            cmd_step.commands << load_workflow_step(step_key, step_inputs)
```

Comment:
the `.commands` is not necessary, the `CommandStep` class should do the right thing when you `<<` on it (taking care of flattening arrays and stuff like that):

```suggestion
            cmd_step << load_workflow_step(step_key, step_inputs)
```

Reply:
Switched - the type from `load_workflow_step` will _now_ only be of `String` or `Array` class(es) - the latter for workdir step translation, but will apply for any others that need to pass in more than just one `command` for concetting